src/lib/modal.js

```diff
@@ -1,24 +1,70 @@
 import { modal_main, modal_background, modal_card, modal_head } from '../styles/modal';
 import { api } from '../utils/platform';
 
-const modal = source =>  {
+const CREDITKEY_MODAL_ID = 'creditkey-modal';
+
+window.creditKey = window.creditKey || {};
+const modalState = window.creditKey.modalState = window.creditKey.modalState || {};
+
+const isModalCreated = () => {
+  return document.getElementById(CREDITKEY_MODAL_ID) !== null;
+};
+
+const displayHiddenModal = () => {
+  const existingModal = document.getElementById(CREDITKEY_MODAL_ID);
+
+  if (existingModal !== null) {
+    existingModal.style.display = 'flex';
+  }
+};
+
+const createAndDisplayModal = () => {
+  // Otherwise, create the modal.
+
   const body = document.body;
   const style = 'margin: auto; width: 100%; border: none; height: 820px;';
-  let iframe = `<iframe id="creditkey-iframe" src="${source + '?modal=true'}" style="${style}"></iframe>`;
+  let iframe = `<iframe id="creditkey-iframe" src="${modalState.source + '?modal=true'}" style="${style}"></iframe>`;
 
-  if (!validate_url(source)) {
+  if (!validate_url(modalState.source)) {
     iframe = `An invalid resource was requested`;
   }
 
   //body.addEventListener('click', e => remove());
-  return body.insertAdjacentHTML('beforeend', `<div id="creditkey-modal" style="${modal_main}"><div style="${modal_background}"></div><div id="modal-card" style="${modal_card}">${iframe}</div></div>`);
-}
+  body.insertAdjacentHTML('beforeend', `<div id="${CREDITKEY_MODAL_ID}" style="${modal_main}"><div style="${modal_background}"></div><div id="modal-card" style="${modal_card}">${iframe}</div></div>`);
+  document.getElementById(CREDITKEY_MODAL_ID).addEventListener('load', function (e) {
+    modalState.index += 1;
+    console.log('new modal state index: %s', modalState.index);
```

Comment:
unless this offers some utility in a production setting, I'd vote to remove any console logs